### PR TITLE
test: optimize memory usage when building storybook by using swc

### PR DESCRIPTION
**Pull request by davidmurdoch**
Building storybook in CI seems to fail often -- probably due to running out of memory. Changing the compiler to swc might help (according to https://github.com/storybookjs/storybook/issues/6408#issuecomment-1655220274).

I don't know what the ramifications of this change may be (I don't even know what storybook is for 😅). So someone with knowledge here should review this PR.

### CODE REVIEW

File: ui/components/ui/icon/icon.stories.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ const IconItem = ({ Component }) => {
       backgroundColor={BackgroundColor.backgroundDefault}
     >
       <Box marginBottom={2}>{Component}</Box>
-      <code>{`${Component.type.__docgenInfo.displayName}`}</code>
+      <code>{`${Component.type.name}`}</code>
     </Card>
```
Comment (georgewrmarshall) [inline ui/components/ui/icon/icon.stories.js:79]: You fixed it!

Screenshot 2024-02-16 at 8 48 04 AM

Screenshot 2024-02-16 at 8 47 50 AM

File: ui/components/ui/logo/logo.stories.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ const LogoItem = ({ Component }) => {
       backgroundColor={BackgroundColor.backgroundDefault}
     >
       <Box marginBottom={2}>{Component}</Box>
-      <Text>{`${Component.type.__docgenInfo.displayName}`}</Text>
+      <Text>{`${Component.type.name}`}</Text>
     </Card>
```
Comment (georgewrmarshall) [inline ui/components/ui/logo/logo.stories.js:49]: Was this just deprecated? Don't see any visual changes

Screenshot 2024-02-16 at 8 49 22 AM

Screenshot 2024-02-16 at 8 49 02 AM

**Review by georgewrmarshall — APPROVED** [review]
LGTM! Thanks for these fixes 🙏

File: ui/components/ui/icon/icon.stories.js (hunk 1 above)
Comment (HowardBraham) [reply ui/components/ui/icon/icon.stories.js:79]: No, this line didn't fix this screen.  The changes in README.mdx fixed this screen.

Comment (georgewrmarshall) [reply ui/components/ui/icon/icon.stories.js:79]: Oh yes that makes sense!

File: ui/components/ui/logo/logo.stories.js (hunk 2 above)
Comment (HowardBraham) [reply ui/components/ui/logo/logo.stories.js:49]: I'm not completely sure, but once @davidmurdoch turned on useSWC, this stopped working without changing `Component.type.__docgenInfo.displayName` to `Component.type.name`.

**Review by HowardBraham — APPROVED** [review]

**Pull request merged**